core/services/ocr2/plugins/ocr2keeper/evm21/mercury/upkeep_failure_reasons.go

```diff
@@ -0,0 +1,15 @@
+package mercury
+
+type MercuryUpkeepFailureReason uint8
+
+const (
+	// upkeep failure onchain reasons
+	MercuryUpkeepFailureReasonNone                    MercuryUpkeepFailureReason = 0
+	MercuryUpkeepFailureReasonTargetCheckReverted     MercuryUpkeepFailureReason = 3
+	MercuryUpkeepFailureReasonUpkeepNotNeeded         MercuryUpkeepFailureReason = 4
+	MercuryUpkeepFailureReasonMercuryCallbackReverted MercuryUpkeepFailureReason = 7
+	// leaving a gap here for more onchain failure reasons in the future
+	// upkeep failure offchain reasons
+	MercuryUpkeepFailureReasonMercuryAccessNotAllowed MercuryUpkeepFailureReason = 32
+	MercuryUpkeepFailureReasonInvalidRevertDataInput  MercuryUpkeepFailureReason = 34
```

Comment:
how can we remove this duplication. This is just adding surface area for bugs when this file goes out of sync with interface file

Reply:
How about we move the states/errors in interface file to a separate (standalone and light) package, so we can share states/errors across different services, mercury is one of the service referencing the states/errors?

Reply:
possibly. looking at interface package, the circular dependency is because of `DecodeStreamsLookupRequest(data []byte) (*mercury.StreamsLookupError, error)` 

Not sure if we can move to some mercury package to remove the cyclic dependency